code/items/base/Ammo.cs

```diff
@@ -72,10 +79,18 @@ private void GiveAmmo( Player player )
 		if ( !player.Inventory.HasWeaponOfAmmoType( Type ) )
 			return;
 
-		player.GiveAmmo( Type, CurrentCount );
+		int iMaxAmmo = Inventory.GetAmmoLimit( Type );
+		int iAdd = Math.Min( CurrentCount, iMaxAmmo - player.AmmoCount( Type ) );
+		if ( iAdd <= 0 ) return;
```

Comment:
we typically just do camelcase for variables and remove any `i` prefix, I'm assuming `i` is for int? Also, use `var`.

Reply:
Keyword values like float, int, bool can be typed out. Use var for anything else.